runtime/onert/backend/train/KernelGenerator.h

```diff
@@ -50,9 +50,11 @@ class KernelGenerator : public backend::train::KernelGeneratorBase
   void visit(const ir::train::operation::ElementwiseActivation &) override;
   void visit(const ir::train::operation::FullyConnected &) override;
   void visit(const ir::train::operation::Loss &) override;
+  void visit(const ir::train::operation::Pool2D &) override;
   void visit(const ir::train::operation::Reshape &node) override;
 
 private:
+  const ir::Operands &_ctx;
```

Comment:
This is necessary to get `pool layer`'s input and output tensor shape.
In/Out shape is used to calculate the pool layer's padding.

Reply:
(Optional) How about using the base member `_tgraph` instead of adding this member (because performance is not important where this member is used) ?

Reply:
Oh, I see. updated!